### PR TITLE
Add configurable prefix to Consumer Group in IngestionJob's Kafka reader

### PR DESCRIPTION
**Pull request by terryyylim**
**What this PR does / why we need it**:
To allow multiple deployments with jobs running in parallel, we should separate their Kafka consumer groups which is based on the JobId. This would prevent overwriting each other's offsets.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: job-controller/src/main/resources/application.yml [hunk 1]
```diff
@@ -23,6 +23,9 @@ feast:
     # Enabling JobManagement
     enabled: true
 
+    # Prefix for JobId
```
Comment (woop) [inline job-controller/src/main/resources/application.yml:26]: This comment says absolutely nothing.

File: job-controller/src/main/java/feast/jobcontroller/runner/ConsolidatedJobStrategy.java [hunk 2]
```diff
@@ -71,6 +74,9 @@ private String createJobId(SourceProto.Source source) {
                 source.getKafkaSourceConfig().getBootstrapServers(),
                 source.getKafkaSourceConfig().getTopic()),
             dateSuffix);
+    if (!this.jobProperties.getJobIdPrefix().isEmpty()) {
```
Comment (woop) [inline job-controller/src/main/java/feast/jobcontroller/runner/ConsolidatedJobStrategy.java:77]: What happens when jobProperties is null?

File: job-controller/src/main/java/feast/jobcontroller/config/JobControllerConfig.java [hunk 3]
```diff
@@ -87,9 +87,9 @@ public JobGroupingStrategy getJobGroupingStrategy(
     Boolean shouldConsolidateJobs =
         feastProperties.getJobs().getController().getConsolidateJobsPerSource();
     if (shouldConsolidateJobs) {
-      return new ConsolidatedJobStrategy(jobRepository);
+      return new ConsolidatedJobStrategy(jobRepository, feastProperties.getJobs());
```
Comment (woop) [inline job-controller/src/main/java/feast/jobcontroller/config/JobControllerConfig.java:90]: Can we extract the jobProperties before passing it? It's a convention that I am trying to get us to follow, even if it takes more lines of code

File: job-controller/src/main/java/feast/jobcontroller/runner/ConsolidatedJobStrategy.java (hunk 2 above)
Comment (terryyylim) [reply job-controller/src/main/java/feast/jobcontroller/runner/ConsolidatedJobStrategy.java]: Addressed case.

File: job-controller/src/main/resources/application.yml (hunk 1 above)
Comment (terryyylim) [reply job-controller/src/main/resources/application.yml]: Updated comment.

File: job-controller/src/main/java/feast/jobcontroller/config/JobControllerConfig.java (hunk 3 above)
Comment (terryyylim) [reply job-controller/src/main/java/feast/jobcontroller/config/JobControllerConfig.java]: Extracted before passing jobProperties.

**Review by pyalex — APPROVED** [review]

**Pull request merged**